lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua

```diff
@@ -91,20 +91,35 @@ function FuelPrices:AddPumpExtensions( pump )
     if SERVER then
         function pump:ChargeCustomer()
             local customer = pump.lastUser
-            local finalPrice = pump:CalculateFuelPrice()
+            local price = pump:CalculateFuelPrice()
 
-            if finalPrice == 0 then return end
+            if price == 0 then return end
 
             -- Ensures we don't charge them more than they have
             local customerMoney = customer:getDarkRPVar( "money" )
-            finalPrice = math.Clamp( finalPrice, 0, customerMoney )
+            price = math.Clamp( finalPrice, 0, customerMoney )
+
+            local priceStruct = {
+                price = price
+            }
+
+            function priceStruct:SetPrice( newPrice )
+                priceStruct.price = newPrice
+            end
+
+            local hookResult = hook.Run( "SimfPhys_FuelTaxes_WillChargeCustomer", pump, customer, priceStruct )
```

Comment:
Might be better as ShouldChargeCustomer, "will" seems weird here

Reply:
It's an event that lets you cancel the default, or modify the price, that's why I went with "Will"

"I will charge the customer with these values, does anyone have something to say about that?"

Reply:
I've not seen any other hook named like this, and hooks like "HUDShouldDraw" act almost exactly like this and use "should"

Reply:
I don't like that I have to be consistent with a gross pattern like this 😒

Reply:
;)